I've tried moving `Windows.h` to the top level, but that fails to compile:
~~~
<module-includes>:2:10: note: in file included from <module-includes>:2:
#include "WinSock2.h"
         ^
C:\Program Files (x86)\Windows Kits\10\/Include/10.0.18362.0/um/WinSock2.h:116:10: note: in file included from C:\Program Files (x86)\Windows Kits\10\/Include/10.0.18362.0/um/WinSock2.h:116:
#include <ws2def.h>
         ^
C:\Program Files (x86)\Windows Kits\10\/Include/10.0.18362.0/shared\ws2def.h:238:16: error: redefinition of 'sockaddr'
typedef struct sockaddr {
               ^
C:\Program Files (x86)\Windows Kits\10\/Include/10.0.18362.0/um\winsock.h:482:8: note: previous definition is here
struct sockaddr {
       ^
<module-includes>:2:10: note: in file included from <module-includes>:2:
#include "WinSock2.h"
         ^
C:\Program Files (x86)\Windows Kits\10\/Include/10.0.18362.0/um/WinSock2.h:116:10: note: in file included from C:\Program Files (x86)\Windows Kits\10\/Include/10.0.18362.0/um/WinSock2.h:116:
#include <ws2def.h>
         ^
C:\Program Files (x86)\Windows Kits\10\/Include/10.0.18362.0/shared\ws2def.h:247:3: error: typedef redefinition with different types ('struct (anonymous struct at C:\Program Files (x86)\Windows Kits\10\/Include/10.0.18362.0/shared\ws2def.h:238:16)' vs 'struct sockaddr')
} SOCKADDR, *PSOCKADDR, FAR *LPSOCKADDR;
  ^
C:\Program Files (x86)\Windows Kits\10\/Include/10.0.18362.0/um\winsock.h:1005:25: note: previous definition is here
typedef struct sockaddr SOCKADDR;
                        ^
~~~

I guess, submodules are not allowed to depend on top level modules?

--- a/stdlib/public/Platform/winsdk.modulemap
+++ b/stdlib/public/Platform/winsdk.modulemap
@@ -337,6 +337,11 @@ module WinSDK [system] {
     link "Gdi32.Lib"
   }
 
+  module WinNT {
+    header "winnt.h"
+    export *
+  }
+
   module WinReg {
     header "winreg.h"
     export *

